packages/ui/src/providers/Config/createClientConfig/fields.tsx

```diff
@@ -40,23 +41,27 @@ function generateFieldPath(parentPath: string, name: string): string {
   return tabPath
 }
 
-export const createClientField = ({
-  clientField,
-  createMappedComponent,
-  field: incomingField,
-  i18n,
-  importMap,
-  parentPath,
-  payload,
-}: {
+export type CreateClientField = (props: {
   clientField: ClientField
   createMappedComponent: CreateMappedComponent
   field: Field
   i18n: I18nClient
   importMap: ImportMap
   parentPath?: string
   payload: Payload
-}): ClientField => {
+  skipFieldComponent?: boolean
+}) => ClientField
+
+export const createClientField: CreateClientField = ({
```

Comment:
We should add the same disclaimer you have in the PR description here as a jsdoc about calling it only if strictly necessary